File: packages/baseline-py/src/baseline_py/baseline/write_baseline.py
"""Write the baseline atomically and deterministically."""

import json
from pathlib import Path

from baseline_py.baseline.baseline_file import BaselineFile


def write_baseline(path: Path, baseline: BaselineFile) -> None:
    """Write the baseline through a temporary file, sorted for a clean diff."""
    document = {
        "schema_version": baseline.schema_version,
        "tool_version": baseline.tool_version,
        "entries": sorted(baseline.entries),
    }
    temporary = path.with_suffix(f"{path.suffix}.tmp")
    temporary.write_text(json.dumps(document, indent=2) + "\n", encoding="utf-8")
    temporary.replace(path)

File: packages/db-quality/src/baseline/writeBaseline.ts
import { renameSync, writeFileSync } from 'node:fs'
import { join } from 'node:path'

import { BASELINE_FILENAME } from '@/baseline/BASELINE_FILENAME.js'
import type { BaselineFile } from '@/baseline/BaselineFile.js'

/** Writes the baseline atomically: a temporary file, then a rename over the target. */
export const writeBaseline = (root: string, baseline: BaselineFile): void => {
  const path = join(root, BASELINE_FILENAME)
  const document = {
    schemaVersion: 1,
    toolVersion: baseline.toolVersion,
    entries: [...baseline.entries].sort(),
  }
  writeFileSync(`${path}.tmp`, `${JSON.stringify(document, null, 2)}\n`)
  renameSync(`${path}.tmp`, path)
}
